File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/module/VideosApi.java
package com.ihui.ai.sdk.module;

import com.fasterxml.jackson.databind.JsonNode;
import com.ihui.ai.sdk.BaseClient;
import com.ihui.ai.sdk.model.VideoGenerationsRequest;

import java.util.Map;

/**
 * 视频模块 — 生成 / 任务查询 / 编排。
 *
 * <p>端点(3 个):
 * <ul>
 *   <li>POST /v1/videos/generations</li>
 *   <li>GET  /v1/videos/tasks/:id</li>
 *   <li>POST /v1/videos/compose</li>
 * </ul>
 */
public final class VideosApi {

    private final BaseClient client;

    /**
     * 构造 VideosApi。
     *
     * @param client 底层 BaseClient
     */
    public VideosApi(BaseClient client) {
        this.client = client;
    }

    /**
     * POST /v1/videos/generations(视频生成,异步任务)。
     *
     * @param req 视频生成请求
     * @return 生成任务(JsonNode,含 taskId / status)
     */
    public JsonNode generations(VideoGenerationsRequest req) {
        return client.request("POST", "/videos/generations", req, JsonNode.class);
    }

    /**
     * GET /v1/videos/tasks/:id(查询视频任务状态)。
     *
     * @param taskId 任务 ID
     * @return 任务状态(JsonNode)
     */
    public JsonNode getTask(String taskId) {
        return client.request("GET", "/videos/tasks/" + BaseClient.encode(taskId),
                null, JsonNode.class);
    }

    /**
     * POST /v1/videos/compose(视频编排)。
     *
     * @param req 编排请求(clips / transitions 等)
     * @return 编排结果(JsonNode)
     */
    public JsonNode compose(Map<String, Object> req) {
        return client.request("POST", "/videos/compose", req, JsonNode.class);
    }
}

File: packages/sdk/dotnet/Modules/VideosApi.cs
using System.Text.Json;

namespace Ihui.AI;

/// <summary>
/// 视频模块 — 生成 / 任务查询 / 编排。
/// </summary>
/// <remarks>
/// 端点(3 个):
/// <list type="bullet">
///   <item>POST /v1/videos/generations</item>
///   <item>GET  /v1/videos/tasks/:id</item>
///   <item>POST /v1/videos/compose</item>
/// </list>
/// </remarks>
public sealed class VideosApi
{
    private readonly BaseClient _client;

    internal VideosApi(BaseClient client) => _client = client;

    /// <summary>POST /v1/videos/generations(视频生成,异步任务)。</summary>
    /// <param name="req">视频生成请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>生成任务(JsonElement,含 taskId / status)</returns>
    public Task<JsonElement?> GenerationsAsync(VideoGenerationsRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/videos/generations", req, cancellationToken);

    /// <summary>GET /v1/videos/tasks/:id(查询视频任务状态)。</summary>
    /// <param name="taskId">任务 ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>任务状态(JsonElement)</returns>
    public Task<JsonElement?> GetTaskAsync(string taskId, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/videos/tasks/" + BaseClient.Encode(taskId), null, cancellationToken);

    /// <summary>POST /v1/videos/compose(视频编排)。</summary>
    /// <param name="req">编排请求(clips / transitions 等)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>编排结果(JsonElement)</returns>
    public Task<JsonElement?> ComposeAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/videos/compose", req, cancellationToken);
}
